src/app/api-server/src/app.rs

```diff
@@ -434,7 +432,9 @@ pub async fn init_dependencies(
     }
 
     if config.dataset_env_vars.encryption_key.as_ref().is_none() {
-        error!("Dataset env vars encryption key was not provided. This feature will be disabled.")
+        error!("Dataset env vars encryption key was not provided. This feature will be disabled.");
```

Comment:
I think because of a deploy configuration error, we shouldn't turn off the feature

Reply:
maybe consider an error and exit?

Reply:
It's not an issue of deployment only. If user doesn't want to configure this he can skip and we should disable it for him

Reply:
A slight problem here I think is that if user wants to actually turn off the feature permanently - they will be getting this error message all the time.

One solution could be a config structure like:
```rust
struct DatasetEnvVarsConfig {
  enabled: Option<bool>,
  encryption_key: Option<String>,
}
```
Then we have 3 situations:
- `enabled: None, encryption_key: None` - write warning that the feature is not configured
- `enabled: false, encryption_key: _` - feature disabled
- `enabled: true, encryption_key: None` - panic
- `enabled: None | true, encryption_key: Some("..")` - feature enabled

What do you think about such 3-state `enabled` logic?

As we introduce new features this will allow us:
- maintain config compatibility - not crash if new features are not configured
- warn users that they are potentially missing out on not configuring new features
- let them silence the warnings to keep a feature disabled